ui/lib/apps/Statement/components/StatementsTable.tsx

```diff
@@ -49,15 +50,17 @@ export default function StatementsTable({
     showFullSQL,
   ])
 
-  function handleRowClick(rec) {
+  const handleRowClick = usePersistFn((rec) => {
     const qs = DetailPage.buildQuery({
       digest: rec.digest,
       schema: rec.schema_name,
       beginTime: timeRange.begin_time,
       endTime: timeRange.end_time,
     })
     navigate(`/statement/detail?${qs}`)
-  }
+  })
+
+  const getKey = useCallback((row) => `${row.digest}_${row.schema_name}`, [])
```

Comment:
Just curious, what's the difference between usePersistFn and useCallback, seem useCallback also works for handleRowClick? (I guess the difference is caused by `navgiate`?)

Reply:
Same: Both return a function that will not be changed.

Difference:

- `a = useCallback(closure, dep)` is persisting the `closure` (and persist again when `dep` is changed), so that the variables captured in `closure` are not changed when `a` is called. This will cause problems if you reference something in the closure but not specified in `dep`. 

- `b = usePersistFn(closure)` is persisting another function that calls `closure` internally. This will not cause problems (reading stall data) when variables captured in `closure` is changed (since the closure is never persisted). `usePersistFn` is an alias to `useEventCallback` in the [React Hook FAQ](https://reactjs.org/docs/hooks-faq.html#how-to-read-an-often-changing-value-from-usecallback).

However as described by the React Hook FAQ, `usePersistFn` pattern is not recommended, so it is only used when `useCallback` is not suitable (i.e. changes too often).